NameSurfer.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * File: NameSurfer.java
+ * ---------------------
+ * When it is finished, this program will implements the viewer for
+ * the baby-name database described in the assignment handout.
+ */
+
+import acm.program.*;
+import java.awt.event.*;
+import javax.swing.*;
+
+public class NameSurfer extends Program implements NameSurferConstants {
+
+/* Instance Variables */
+	private JTextField tf;
+	private JButton graph;
+	private JButton clear;
+	private NameSurferDataBase repo;
+	private NameSurferGraph guiGraph;
+	
+/* Method: init() */
+/**
+ * This method has the responsibility for reading in the data base
+ * and initializing the interactors at the bottom of the window.
+ */
+	public void init() {
+		// Initializing interactors and creating NameSurferGraph instance
+		guiGraph = new NameSurferGraph();
+		add(guiGraph);
+		add(new JLabel("Name "), SOUTH);
+	    tf = new JTextField(TEXT_FIELD_SIZE);
+	    tf.addActionListener(this);
+	    add(tf, SOUTH);
+		add(graph = new JButton("Graph"), SOUTH);
+		add(clear = new JButton("Clear"), SOUTH);
+	    addActionListeners();
+	    // Creating new instance of NameSurferDatabase with file read-in
+	    repo = new NameSurferDataBase(NAMES_DATA_FILE);
+	}
+
+/* Method: actionPerformed(e) */
+/**
+ * This class is responsible for detecting when the buttons are
+ * clicked, so you will have to define a method to respond to
+ * button actions.
+ */
+	public void actionPerformed(ActionEvent e) {
+		if(e.getSource() == tf) {
+			// Draws graph of name entered into textfield:
+			// Add entry to arraylist and then pull data from arraylist to be displayed
+			guiGraph.addEntry(repo.findEntry(tf.getText()));
```

Comment:
Need to handle the null value when the name is not found. Will cause a crash.

Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at NameSurferGraph.drawGraphLines(NameSurferGraph.java:67)
	at NameSurferGraph.update(NameSurferGraph.java:57)
	at NameSurfer.actionPerformed(NameSurfer.java:52)
	at javax.swing.JTextField.fireActionPerformed(JTextField.java:508)
	at javax.swing.JTextField.postActionEvent(JTextField.java:721)
	at javax.swing.JTextField$NotifyAction.actionPerformed(JTextField.java:836)
	at javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1662)
	at javax.swing.JComponent.processKeyBinding(JComponent.java:2879)
	at javax.swing.JComponent.processKeyBindings(JComponent.java:2926)
	at javax.swing.JComponent.processKeyEvent(JComponent.java:2842)
	at java.awt.Component.processEvent(Component.java:6293)
	at java.awt.Container.processEvent(Container.java:2229)
	at java.awt.Component.dispatchEventImpl(Component.java:4872)
	at java.awt.Container.dispatchEventImpl(Container.java:2287)
	at java.awt.Component.dispatchEvent(Component.java:4698)
	at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1887)
	at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:762)
	at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1027)
	at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:899)
	at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:727)
	at java.awt.Component.dispatchEventImpl(Component.java:4742)
	at java.awt.Container.dispatchEventImpl(Container.java:2287)
	at java.awt.Window.dispatchEventImpl(Window.java:2719)
	at java.awt.Component.dispatchEvent(Component.java:4698)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:747)
	at java.awt.EventQueue.access$300(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:706)
	at java.awt.EventQueue$3.run(EventQueue.java:704)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.awt.EventQueue$4.run(EventQueue.java:718)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:717)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)